### PR TITLE
Problem: Validator APY and uptime additions

### PR DESCRIPTION
**Pull request by cdc-Hitesh**
image

### CODE REVIEW

**Review by crypto-matto — COMMENTED** [review]
1. The APY value is between 0 and 1, which we shall multiply the value 100 to get the APY in %
2. We may add sorting to `APY` and `Uptime` field

File: src/pages/staking/staking.tsx [hunk 1]
```diff
@@ -165,39 +166,48 @@ const FormDelegationRequest = () => {
...
+              CUMULATIVE_SHARE_PERCENTAGE_THRESHOLD,
+            ) &&
+            !displayedWarningColumn
+          ) {
+            displayedWarningColumn = true;
+            willDisplayWarningColumn = true;
+          }
+          const [validatorApyRaw, validatorUptime] = await Promise.all([
+            apiClient.getValidatorsAverageApy([validator.validatorAddress]),
+            apiClient.getValidatorUptimeByAddress(validator.validatorAddress),
```
Comment (crypto-matto) [inline src/pages/staking/staking.tsx:188]:
```suggestion
            apiClient.getValidatorUptimeByAddress(validator.validatorAddress),
```

Definitely not a good idea to call API client hundred times here. 
Can we use a `getValidatorsUptime()` instead, which could return uptimes of an input array of validatorsAddressList?


Screenshot 2021-11-02 at 5 12 27 PM

File: src/pages/staking/staking.tsx [hunk 2]
```diff
@@ -165,39 +166,48 @@ const FormDelegationRequest = () => {
 
   const [t] = useTranslation();
 
-  const processValidatorList = (validatorList: ValidatorModel[] | null) => {
+  const processValidatorList = async (validatorList: ValidatorModel[] | null) => {
     if (validatorList) {
       let willDisplayWarningColumn = false;
       let displayedWarningColumn = false;
+      const apiClient = ChainIndexingAPI.init(currentSession.wallet.config.indexingUrl);
+
```
Comment (crypto-matto) [inline src/pages/staking/staking.tsx:174]: We shall call here
```
            apiClient.getValidatorsAverageApy([validator.validatorAddressList]),
            apiClient.getValidatorUptimeByAddress([validator.validatorAddressList]),
```

**Review by crypto-matto — CHANGES_REQUESTED** [review]
```

```
Screenshot 2021-11-02 at 5 12 27 PM

**Comment by crypto-matto** [comment]
Is it possible to fetch the APY & uptime in one batch from `NodeRpcService.tsx` => `loadTopValidators()`?

**Review by crypto-matto — APPROVED** [review]

**Review by XinyuCRO — APPROVED** [review]

File: src/pages/staking/staking.tsx [hunk 3]
```diff
@@ -165,66 +161,30 @@ const FormDelegationRequest = () => {
...
+    // const moderationConfigHandler = async () => {
+    //   try {
+    //     const fetchModerationConfigData = await fetch(MODERATION_CONFIG_FILE_URL);
+    //     const moderationConfigData = await fetchModerationConfigData.json();
+    //     setModerationConfig(moderationConfigData);
+    //   } catch (error) {
+    //     // eslint-disable-next-line no-console
+    //     console.error('Error occurred while fetching moderation config file', error);
+    //   }
+    // };
```
Comment (crypto-eddy) [inline src/pages/staking/staking.tsx:179]: This can just be cleaned if not needed anymore

Comment (cdc-Hitesh) [reply src/pages/staking/staking.tsx:179]: @crypto-matto  I think we can discard this PR right?

Comment (crypto-matto) [reply src/pages/staking/staking.tsx:179]: Let me double check on this.

**Comment by crypto-matto** [comment]
The validator list section is crashing after merged with latest dev branch. I'm investigating why. Don't merge it yet.

**Comment by crypto-matto** [comment]
> The validator list section is crashing after merged with latest dev branch. I'm investigating why. Don't merge it yet.

Problem solved. It's due to the deprecated `WalletService.ts`

@cdc-Hitesh Is `WalletService.ts` gonna wipe out completely? We may create an extra PR to fix this.

**Review by crypto-matto — APPROVED** [review]

**Review by XinyuCRO — APPROVED** [review]

**Review by crypto-eddy — APPROVED** [review]

**Comment by crypto-eddy** [comment]
> > The validator list section is crashing after merged with latest dev branch. I'm investigating why. Don't merge it yet.
> 
> Problem solved. It's due to the deprecated `WalletService.ts`. Another PR regarding to this issue #813 will be addressed separately.

Has this been fixed by now ?

**Comment by crypto-matto** [comment]
> #813

Yes, this issue has been closed. 

> > > The validator list section is crashing after merged with latest dev branch. I'm investigating why. Don't merge it yet.
> > 
> > 
> > Problem solved. It's due to the deprecated `WalletService.ts`. Another PR regarding to this issue #813 will be addressed separately.
> 
> Has this been fixed by now ?

Yes. Issue #813 was closed. The `dev` branch is fine. This only happens on this `feature/#748/validatorList`, and the deprecated functions in `WalletService` are removed in commit [#34427cf](https://github.com/crypto-com/chain-desktop-wallet/pull/781/commits/34427cfc0817d06e6a3fd0798c1c2f36b51c7378)

**Review by crypto-matto — APPROVED** [review]

**Pull request merged**